src/MSBuild/XMake.cs

```diff
@@ -124,7 +124,9 @@ public enum ExitType
         /// <summary>
         /// Static constructor
         /// </summary>
+#pragma warning disable CA1810 // Initialize reference type static fields inline
```

Comment:
Why is this one problematic?

Reply:
It is initializing static fields here. But there is a big block comment:
```
////////////////////////////////////////////////////////////////////////////////
//  Only initialize static fields here, not inline!                           //
//  This forces the type to initialize in this static constructor and thus    //
//  any configuration file exceptions can be caught here.                     //
////////////////////////////////////////////////////////////////////////////////
```
So I put a suppression in.

Reply:
Same for this one

Reply:
Static ctors are invoked automatically by the runtime, yes. `MSBuildApp` is the entry point so it's not clear why we wouldn't want to do this work at the top of `Main` instead. It should be equivalent unless someone directly calls into other `MSBuildApp` methods (unlikely).